support/metrics/metricsTracker.go

```diff
@@ -286,3 +350,36 @@ func (mt *MetricsTracker) sendEvent(eventType string, eventProps interface{}) er
 	}
 	return nil
 }
+
+func mergeEventProps(commonProps, eventProps map[string]interface{}) (map[string]interface{}, error) {
+	m := make(map[string]interface{})
+	for k, v := range commonProps {
+		m[k] = v
+	}
+
+	for k, v := range eventProps {
+		m[k] = v
+	}
+
+	return m, nil
+}
+
+func toMapStringInterface(v interface{}) (map[string]interface{}, error) {
```

Comment:
this seems like a generic enough function to where we can put this in `utils.functions.go` - along with a comment describing what it does since it deals with a lot of `interface{}` types so is hard to tell what it does without reading the code. quick one-line comment will help - can use an example to help if you want.